File: jimmer-ddd-java-spring-boot-starter/src/test/java/org/morecup/jimmerddd/java/spring/domain/SnowflakeIdGenerator.java
package org.morecup.jimmerddd.java.spring.domain;


import cn.hutool.core.lang.Snowflake;
import cn.hutool.core.util.IdUtil;
import org.babyfish.jimmer.sql.meta.UserIdGenerator;

public class SnowflakeIdGenerator implements UserIdGenerator<Long> {

    // 静态成员变量，对应 Kotlin 的 companion object
    private static final Snowflake snowflake = IdUtil.getSnowflake(1, 1);

    @Override
    public Long generate(Class<?> entityType) {
        return snowflake.nextId();
    }
}

File: jimmer-ddd-kotlin-spring-boot-starter/src/test/kotlin/org/morecup/jimmerddd/kotlin/spring/domain/SnowflakeIdGenerator.kt
package org.morecup.jimmerddd.kotlin.spring.domain

import cn.hutool.core.lang.Snowflake
import cn.hutool.core.util.IdUtil
import org.babyfish.jimmer.sql.meta.UserIdGenerator

class SnowflakeIdGenerator : UserIdGenerator<Long> {
    companion object {
        val snowflake: Snowflake = IdUtil.getSnowflake(1, 1)
    }

    override fun generate(entityType: Class<*>): Long? {
        return snowflake.nextId()
    }
}